packages/web-app-files/tests/unit/components/SideBar/Links/PublicLinks/LinkEdit.spec.js

```diff
@@ -0,0 +1,618 @@
+    })
+  })
+
+  describe('link password delete button', () => {
+    it('should be present if password is not enforced and the link has a password set', () => {
+      const wrapper = getShallowMountedWrapper(
+        createStore({
+          linkInEdit: {
+            hasPassword: true
+          }
+        })
+      )
+      const passwordDeleteButtonElement = wrapper.find(selectors.linkPasswordDeleteButton)
+
+      expect(passwordDeleteButtonElement.exists()).toBeTruthy()
+    })
+
+    it('should not be present if password is not enforced', () => {
+      const wrapper = getShallowMountedWrapper()
+      const passwordDeleteButtonElement = wrapper.find(selectors.linkPasswordDeleteButton)
+
+      expect(passwordDeleteButtonElement.exists()).toBeFalsy()
+    })
+    it('should not be present if the link does not has a password set', () => {
+      const wrapper = getShallowMountedWrapper(
+        createStore({
+          linkInEdit: {
+            hasPassword: false
+          }
+        })
+      )
+      const passwordDeleteButtonElement = wrapper.find(selectors.linkPasswordDeleteButton)
+
+      expect(passwordDeleteButtonElement.exists()).toBeFalsy()
+    })
+
+    it('should be present if password is not enforced and password value is set', async () => {
+      const wrapper = getMountedWrapper(
+        createStore({
+          linkInEdit: {}
+        })
+      )
+      const linkPasswordInputElement = wrapper.find(selectors.linkPasswordField)
+      await linkPasswordInputElement.setValue('VeryStrongPassword')
+
+      const passwordDeleteButtonElement = wrapper.find(selectors.linkPasswordDeleteButton)
+
+      expect(passwordDeleteButtonElement.exists()).toBeTruthy()
+    })
+
+    it('should remove password if clicked', async () => {
+      const wrapper = getMountedWrapper(
+        createStore({
+          linkInEdit: {}
+        })
+      )
+      const linkPasswordInputElement = wrapper.find(selectors.linkPasswordField)
+      await linkPasswordInputElement.setValue('VeryStrongPassword')
+
+      const passwordDeleteButtonElement = wrapper.find(selectors.linkPasswordDeleteButton)
+      await passwordDeleteButtonElement.trigger('click')
+
+      expect(wrapper.vm.password).toBe('')
+      expect(wrapper.vm.hasPassword).toBeFalsy()
+    })
+  })
+
+  describe('link edit form validation', () => {
+    it('should be invalid if the expiration date value is not valid', () => {
+      const wrapper = getMountedWrapper(
+        createStore({
+          linkInEdit: {
+            id: 1224,
+            name: 'Public Link',
+            hasPassword: true,
+            // for an enforced expireDate, empty expireDate is an invalid value
+            expireDate: ''
+          },
+          publicLinkCapabilities: getLinkCapabilities({ enforcedExpireDate: true })
+        }),
+        {
+          saving: false
+        }
+      )
+
+      expect(wrapper.vm.$_isValid).toBeFalsy()
```

Comment:
what about checking the `disabled` property of the `oc-button` instead?
That is really what we are interested in

Reply:
fixed

Reply:
I would suggest to go even a step further and not check `$_isValid` at all. We don't care what this "private" function returns, we are only interested if the `disabled` attribute is set correctly. So I feel like deleting the above line.
Same it other similar checks